<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1440</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>30</y>
      <width>811</width>
      <height>661</height>
     </rect>
    </property>
    <property name="dragMode">
     <enum>QGraphicsView::RubberBandDrag</enum>
    </property>
    <property name="transformationAnchor">
     <enum>QGraphicsView::AnchorUnderMouse</enum>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>541</width>
      <height>263</height>
     </rect>
    </property>
    <property name="title">
     <string>Построить линию</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>1-ая точка</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Х</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="firstXDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="minimum">
             <double>-100000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Y</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="firstYDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="minimum">
             <double>-1000000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::DefaultStepType</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>2-ая точка</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Х</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="secondXDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Y</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="secondYDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="drawLinePushButton">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>280</y>
      <width>541</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>Построить спектр</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>Центр</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Х</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spectreCenterXDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="minimum">
             <double>-100000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Y</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spectreCenterYDoubleSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="minimum">
             <double>-1000000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000000.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::DefaultStepType</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="title">
        <string>Параметры</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Угол</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="spectreAngleDoubleSpinBox">
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="minimum">
           <double>-100000000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000000000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Длина</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="spectreLenDoubleSpinBox">
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="minimum">
           <double>-100000000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000000000.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="drawSpectrePushButton">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>550</y>
      <width>531</width>
      <height>421</height>
     </rect>
    </property>
    <property name="title">
     <string>Параметры</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Алгоритм построения</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QComboBox" name="drawAlgoComboBox">
          <property name="styleSheet">
           <string notr="true">subcontrol-position: right;</string>
          </property>
          <item>
           <property name="text">
            <string>Библиотечный</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>ЦДА</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Брезенхэм (целые)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Брезенхэм (вещественные)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Ву</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_9">
       <property name="title">
        <string>Цвет линий</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Текущий цвет</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="colorFrame">
            <property name="styleSheet">
             <string notr="true">background: black;</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="pickColorPushButton">
          <property name="text">
           <string>Изменить цвет</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_10">
       <property name="title">
        <string>Цвет фона</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QPushButton" name="pickBackgroundColorPushButton">
          <property name="text">
           <string>Изменить цвет</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1440</width>
     <height>31</height>
    </rect>
   </property>
   <widget class="QMenu" name="infoMenu">
    <property name="title">
     <string>Справка</string>
    </property>
   </widget>
   <widget class="QMenu" name="editMenu">
    <property name="title">
     <string>Редактирование</string>
    </property>
    <addaction name="undoAction"/>
   </widget>
   <addaction name="infoMenu"/>
   <addaction name="editMenu"/>
  </widget>
  <action name="undoAction">
   <property name="text">
    <string>Отмена</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
